library/core/src/main/java/com/google/android/exoplayer2/extractor/flv/FlvExtractor.java

```diff
@@ -110,6 +110,7 @@ public boolean sniff(ExtractorInput input) throws IOException, InterruptedExcept
     input.peekFully(scratch.data, 0, 4);
     scratch.setPosition(0);
     int dataOffset = scratch.readInt();
+    if (dataOffset > 9) dataOffset = 9;
```

Comment:
Ok, finding out more sounds good. I'm not sure we have enough information to know what the best fix (or otherwise) is at this point. For example, if the files always have some specific value and fail to play anyway, then it would be much better to check for that value specifically and just return false.